src/huggingface_hub/utils/tqdm.py

```diff
@@ -66,58 +66,124 @@
 from ..constants import HF_HUB_DISABLE_PROGRESS_BARS
 
 
-# `HF_HUB_DISABLE_PROGRESS_BARS` is `Optional[bool]` while `_hf_hub_progress_bars_disabled`
-# is a `bool`. If `HF_HUB_DISABLE_PROGRESS_BARS` is set to True or False, it has priority.
-# If `HF_HUB_DISABLE_PROGRESS_BARS` is None, it means the user have not set the
-# environment variable and is free to enable/disable progress bars programmatically.
-# TL;DR: env variable has priority over code.
+# The `HF_HUB_DISABLE_PROGRESS_BARS` environment variable can be True, False, or not set (None),
+# allowing for control over progress bar visibility. When set, this variable takes precedence
+# over programmatic settings, dictating whether progress bars should be shown or hidden globally.
+# Essentially, the environment variable's setting overrides any code-based configurations.
 #
-# By default, progress bars are enabled.
-_hf_hub_progress_bars_disabled: bool = HF_HUB_DISABLE_PROGRESS_BARS or False
+# If `HF_HUB_DISABLE_PROGRESS_BARS` is not defined (None), it implies that users can manage
+# progress bar visibility through code. By default, progress bars are turned on.
 
 
-def disable_progress_bars() -> None:
+progress_bar_states = {}
```

Comment:
An alternative approach would be to have a flat dictionary instead (typed as `Dict[str, bool]`. So something like:

```py
{"foo.bar": False}
```

When running `enable_progress_bars("foo.bar.something")`, it becomes

```py
{"foo.bar": False, "foo.bar.something": True}
```

Then if you run `disable_progress_bars("foo")`, it becomes:
```py
{"foo": False}
```

---

The logic to do that would be:
- for `disable_progress_bars(name)`: loop through all entries. If `key.startswith(f"{name}.")` => delete the entry. Then set `states[name]` to False
- for `enable_progress_bars(name)`: loop through all entries. If `key.startswith(f"{name}.")`  => delete the entry. Then set `states[name]` to True
- for `is_progress_bars_enabled(name)`: check for `name` => if present, return value. If not, check for "name minus the last part". Then "name minus the 2 last parts", etc until you reach `""`.

WDYT?

Reply:
Thank you so much for your inputs @Wauplin! 🙏 The suggestion to have a flat dict made it easier for me to grasp! I now also removed the private functions and implemented the simplified logic right in the corresponding functions. I think this might work now. 😄